patches/tModLoader/Terraria/Social/Steam/WorkshopHelper.TML.cs

```diff
@@ -314,8 +321,12 @@ internal List<ModDownloadItem> QueryItemsSynchronously(out List<string> missingM
 
 						for (int j = 0; j < _queryReturnCount; j++) {
 							var itemsIndex = j + i * Constants.kNumUGCResultsPerPage;
-							var item = GenerateModDownloadItemFromQuery((uint)j);
-							if (item is null) {
+							var match = TryGenerateModDownloadItem((uint)j, out var item);
+							if (match != WorkshopSearchReturnState.Success) {
+								if (match == WorkshopSearchReturnState.Banned) {
+									throw new BannedModException($"The mod {item.DisplayName} with ID {item.PublishId} is Banned!", item.DisplayName, item.PublishId.ToString());
+								}
+
```

Comment:
Is throwing an exception going to cause problems if the query accidentally matches a banned mod? This method previously didn't throw. Is this one of the cases where a mod should actually be skipped if banned, or is it a case where an item should be returned with `Banned: true`? Needs caller analysis.

In addition, the comment at the top of this method says
```
			/// Note that the List size is 1 to 1 with the provided array.
			/// If the Mod is not found, the space is filled with a null.
```

Which appears to be outdated, probably from when we added the `out List<string> missingMods`